src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -106,6 +107,43 @@ def disable_attention_slicing(self):
         # set slice_size = `None` to disable `attention slicing`
         self.enable_attention_slicing(None)
 
+    @torch.no_grad()
+    def decode_latents(self, latents: torch.FloatTensor) -> np.ndarray:
```

Comment:
Since we are just sending the latents to the callback I think we could return the code inside this function back where it was inside `__call__`.

Reply:
but then the callback wouldn't be able to reference this method, and would have to copy-and-paste it if it wants that very common functionality.

(I'm okay with saying that's out of scope for this PR and will be addressed separately as part of https://github.com/huggingface/diffusers/issues/551#issuecomment-1259415393 )

Reply:
Hi @pcuenca, @keturn makes a good point here. Should we return `decode_latents` and `run_safety_checker` back inside `__call__` then?

Reply:
I think we should for now, there's another discussion in another issue about doing some code extraction of some sort, so I think we should just focus on the bare callback functionality here.

Reply:
Okay, got it!